js/src/modals/AddContract/addContract.js

```diff
@@ -244,11 +247,26 @@ export default class AddContract extends Component {
         this.onClose();
       })
       .catch((error) => {
-        newError(error);
+        this.props.newError(error);
       });
   }
 
   onClose = () => {
     this.props.onClose();
   }
 }
+
+function mapStateToProps (state) {
+  return {};
+}
+
+function mapDispatchToProps (dispatch) {
+  return bindActionCreators({
+    newError
+  }, dispatch);
+}
+
+export default connect(
+  mapStateToProps,
```

Comment:
This is just an optimisation:

> If you omit [`mapDispatchToProps`], the component will not be subscribed to the Redux store.

You could pass `null`, it occurs multiple times in the codebase already.